core/src/main/scala/filodb.core/memstore/OnDemandPagingShard.scala

```diff
@@ -102,44 +102,45 @@ TimeSeriesShard(dataset, storeConfig, shardNum, rawStore, metastore, evictionPol
                          else BinaryRegionLarge.asNewByteArray(bytes, offset)
       partKeyBytesToPage += partKeyBytes
       methods += chunkMethod
       shardStats.partitionsRestored.increment
     }).executeOn(ingestSched)
-  }
+  // No need to execute the task on ingestion thread if it's empty / no ODP partitions
+  } else Task.now(Nil)
 
   /**
    * Creates a Task which is meant ONLY TO RUN ON INGESTION THREAD
    * to create TSPartitions for partIDs found in Lucene but not in in-memory data structures
    * It runs in ingestion thread so it can correctly verify which ones to actually create or not
    */
   private def createODPPartitionsTask(partIDs: Buffer[Int], callback: (Array[Byte], Int) => Unit):
-  Task[Seq[TimeSeriesPartition]] =
-    if (partIDs.nonEmpty) Task {
-      partIDs.map { id =>
-        // for each partID: look up in partitions
-        partitions.get(id) match {
-          case TimeSeriesShard.OutOfMemPartition =>
-            logger.debug(s"Creating TSPartition for ODP from part ID $id")
-            // If not there, then look up in Lucene and get the details
-            for { partKeyBytesRef <- partKeyIndex.partKeyFromPartId(id)
-                  unsafeKeyOffset = PartKeyLuceneIndex.bytesRefToUnsafeOffset(partKeyBytesRef.offset)
-                  group = partKeyGroup(partKeyBytesRef.bytes, unsafeKeyOffset)
-                  part <- Option(createNewPartition(partKeyBytesRef.bytes, unsafeKeyOffset, group, 4)) } yield {
-              partSet.add(part)
-              callback(partKeyBytesRef.bytes, unsafeKeyOffset)
-              part
-            }
-            // create the partition and update data structures (but no need to add to Lucene!)
-            // NOTE: if no memory, then no partition!
-          case p: TimeSeriesPartition => Some(p)
-        }
-      }.toVector.flatten
-    } else Task.now(Nil)
+  Task[Seq[TimeSeriesPartition]] = Task {
+    partIDs.map { id =>
+      // for each partID: look up in partitions
+      partitions.get(id) match {
+        case TimeSeriesShard.OutOfMemPartition =>
+          logger.debug(s"Creating TSPartition for ODP from part ID $id")
+          // If not there, then look up in Lucene and get the details
+          for { partKeyBytesRef <- partKeyIndex.partKeyFromPartId(id)
+                unsafeKeyOffset = PartKeyLuceneIndex.bytesRefToUnsafeOffset(partKeyBytesRef.offset)
+                group = partKeyGroup(partKeyBytesRef.bytes, unsafeKeyOffset)
+                part <- Option(createNewPartition(partKeyBytesRef.bytes, unsafeKeyOffset, group, 4)) } yield {
+            partSet.add(part)
+            callback(partKeyBytesRef.bytes, unsafeKeyOffset)
+            part
+          }
+          // create the partition and update data structures (but no need to add to Lucene!)
+          // NOTE: if no memory, then no partition!
+        case p: TimeSeriesPartition => Some(p)
+      }
+    }.toVector.flatten
+  }
```

Comment:
Can we add a warning message if partIDs.isEmpty ? Bugs like this in the future will be brought to surface.

Reply:
I did better than a warning, I added a require that partIDs cannot be empty  :)